backend/configuration/configure_appwrite.py

```diff
@@ -0,0 +1,539 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2021 Felix Steinkohl <steinkohl@campus.tu-berlin.de>
+
+import os
+from dataclasses import dataclass
+import pandas as pd
+from appwrite.client import Client
+from appwrite.services.users import Users
+from appwrite.services.teams import Teams
+from appwrite.services.database import Database
+from appwrite.exception import AppwriteException
+
+
+@dataclass
+class UserData:
+    """
+    Dataclass for specifying user data
+    """
```

Comment:
Well, no - on this point I disagree with you. At the time this was created, i thought we would use an automatic documentation tool like sphinx, then at least this minimal description would be listed in the docs. And btw - you approved this [PR](https://github.com/amosproj/amos2021ws07-nft-development/pull/79) - hehe :D

Reply:
Ought to be a good advice. This comment is an archetype for what is a reason for being dismissed as job candidate in a coding job interview.

For review approval, I started to ignore code quality for our project  – only commenting on bigger violations, and only the first one of for each kind for Dominic's sake – because our scrum situation is too much stress.

Comments and generated documentation is different. Generation doesn't get into the way of a code reader. A tool which would generate such useless documentation comments is a bad tool. A well-chosen function signature is more expressive then.

Structuring your code into functions helped me the most. The "main function" could need more splitting.

Frankly, after reading the first few doc comments, I entirely ignored all of them. They only wasted time with scrolling. They didn't help me understand the meaning when I didn't immediately understood it. In this file, a good expressive function name always would have been sufficient to get the meaning of any. (I think, some function names are misleading.)

Most of the professional programming time goes into reading and *scrolling*, less since IDEs became more intelligent. But you cannot expect a full-blown IDE everywhere where people get to read it, like on Github. Worse are comments which people need to skip without use.

Comments are not evil but they are easily misused.